pennylane/beta/interfaces/tf.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the mixin interface class for creating differentiable quantum tapes with
+TensorFlow.
+"""
+# pylint: disable=protected-access, attribute-defined-outside-init
+import numpy as np
+import tensorflow as tf
+
+try:
+    from tensorflow.python.eager.tape import should_record_backprop
+except ImportError:
+    from tensorflow.python.eager.tape import should_record as should_record_backprop
+
+
+from pennylane.beta.queuing import AnnotatedQueue
+
+
+class TFInterface(AnnotatedQueue):
+    """Mixin class for applying an TensorFlow interface to a :class:`~.QuantumTape`.
+
+    TensorFlow-compatible quantum tape classes can be created via subclassing:
+
+    .. code-block:: python
+
+        class MyAutogradQuantumTape(AutogradInterface, QuantumTape):
+
+    Alternatively, the autograd interface can be dynamically applied to existing
+    quantum tapes via the :meth:`~.apply` class method. This modifies the
+    tape **in place**.
+
+    Once created, the autograd interface can be used to perform quantum-classical
+    differentiable programming.
+
+    .. note::
+
+        If using a device that supports native autograd computation and backpropagation, such as
+        :class:`~.DefaultQubitTF`, the TensorFlow interface **does not need to be applied**. It is
+        only applied to tapes executed on non-TensorFlow compatible devices.
+
+    **Example**
+
+    One a TensorFlow quantum tape has been created, it can be differentiated using the gradient tape:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=1)
+        p = tf.Variable([0.1, 0.2, 0.3], dtype=tf.float64)
+
+        with tf.GradientTape() as tape:
+            with TFInterface.apply(QuantumTape()) as qtape:
+                qml.Rot(p[0], p[1] ** 2 + p[0] * p[2], p[1] * tf.sin(p[2]), wires=0)
+                expval(qml.PauliX(0))
+
+            result = qtape.execute(dev)
+
+    >>> print(result)
+    tf.Tensor([0.06982072], shape=(1,), dtype=float64)
+    >>> grad = tape.gradient(result, p)
+    >>> print(grad)
+    tf.Tensor([0.29874274 0.39710271 0.09958091], shape=(3,), dtype=float64)

```

Comment:
```suggestion
    tf.Tensor([0.29874274, 0.39710271, 0.09958091], shape=(3,), dtype=float64)
```
? (not 100% sure about this one)

Reply:
I copied and pasted this from terminal output, so I think this is correct.

I think `print()` causes the commas to disappear

Reply:
yep
![image](https://user-images.githubusercontent.com/2959003/93421582-e53c5f80-f8f0-11ea-857f-cd406d59ca29.png)